ios/FluentUI/Divider/Divider.swift

```diff
@@ -0,0 +1,137 @@
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+
+    /// The object that groups properties that allow control over the Divider appearance.
+    @objc open var state: MSFDividerState {
+        return divider.state
+    }
+
+    /// Creates a new MSFDivider instance.
+    ///  - Parameters:
+    ///   - style: The DividerStyle value used by the Divider.
+    ///   - orientation: The DividerOrientation used by the Divider.
+    ///   - spacing: The DividerSpacing used by the Divider.
+    @objc public init(style: MSFDividerStyle = .default, orientation: MSFDividerOrientation = .horizontal, spacing: MSFDividerSpacing = .none) {
+        super.init()
+
+        divider = FluentDivider(style: style, orientation: orientation, spacing: spacing)
+        hostingController = FluentUIHostingController(rootView: AnyView(divider.windowProvider(self)))
+    }
+
+    var window: UIWindow? {
+        return self.window
+    }
+
+    private var hostingController: FluentUIHostingController!
+
+    private var divider: FluentDivider!
+}
+
+/// Properties that can be used to customize the appearance of the Divider.
+@objc public protocol MSFDividerState {
+    /// Sets a custom color for the Divider.
+    var color: UIColor? {get set}
+
+    /// Defines the orientation of the Divider.
+    var orientation: MSFDividerOrientation {get set}
+
+    /// Defines the style of the Divider.
+    var style: MSFDividerStyle {get set}
+
+    /// Defines the spacing of the Divider.
+    var spacing: MSFDividerSpacing {get set}
+}
+
+/// View that represents the Divider.
+public struct FluentDivider: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var state: MSFDividerStateImpl
+    @ObservedObject var tokens: MSFDividerTokens
+
+    /// Creates and initializes a SwiftUI Divider
+    /// - Parameters:
+    ///   - style: The DividerStyle value used by the Divider.
+    ///   - orientation: The DividerOrientation used by the Divider.
+    ///   - spacing: The DividerSpacing used by the Divider.
+    public init (style: MSFDividerStyle = .default, orientation: MSFDividerOrientation = .horizontal, spacing: MSFDividerSpacing = .none) {
+        let state = MSFDividerStateImpl(style: style, orientation: orientation, spacing: spacing)
+
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        @ViewBuilder
+        var dividerBackground: some View {
+            Rectangle()
+                .fill(Color(state.color ?? tokens.color))
+        }
+
+        @ViewBuilder
+        var divider: some View {
+            switch state.orientation {
+            case .horizontal:
+                dividerBackground.frame(height: thickness)
+                    .padding(EdgeInsets(top: tokens.padding, leading: 0, bottom: tokens.padding, trailing: 0))
```

Comment:
nit: do you mind breaking the 4 parameters into multiple lines for readability? (here and line 97)